service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -728,6 +732,76 @@ INSERT INTO resource (workspace_id, cloud_platform, resource_id, name, descripti
     }
   }
 
+  @WriteTransaction
+  public boolean deleteResourceProperties(
+      UUID workspaceUuid, UUID resourceUuid, List<String> propertyKeys) {
+    if (propertyKeys.isEmpty()) {
```

Comment:
i have it here because that's what workspaceDao does. And there's no such validatation in the controller/service yet. I think it makes sense in the controller like auth, but if I move this specific code there in this PR it wouldn't be consistent with the current implementation.

Reply:
I think workspace shouldn't validate in dao also, though that doesn't need to change in this PR.

@ddietterich any thoughts on where request validation should go? Dao seems like the wrong place. Seems like should be controller or service.

Reply:
OK. I will keep this in here that throws 500 (due to bad mutation or something) and throw a 400 in controller.